packages/react/src/hooks/useCogsMediaConfig.ts

```diff
@@ -0,0 +1,23 @@
+import { CogsConnection, MediaClientConfig } from '@clockworkdog/cogs-client';
+import { useEffect, useState } from 'react';
+
+export default function useCogsMediaConfig<
+  MediaConfigExtra extends unknown = Record<never, never>,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+>(connection: CogsConnection<any>): (MediaClientConfig & MediaConfigExtra) | null {
+  const [mediaConfig, setMediaConfig] = useState<(MediaClientConfig & MediaConfigExtra) | null>(null);
+
+  useEffect(() => {
+    // Use the latest config in case it has changed before this useEffect ran
```

Comment:
This pattern ensures that we don't miss any updates due to a race condition where the update comes in before this `useEffect` runs.